### PR TITLE
Davor/product sort

### PR DESCRIPTION
**Pull request by DGjorgjievski**

### CODE REVIEW

File: src/MegaMall.Api/Controllers/ProductController.cs [hunk 1]
```diff
@@ -19,19 +19,20 @@ public ProductController(IProductService service)
 
         [HttpGet]
         [ProducesResponseType(200), ProducesResponseType(500)]
-        public async Task<ActionResult<List<Product>>> GetProducts(int? page, int? take)
-        {
+        public async Task<ActionResult<List<Product>>> GetProducts(int? page, int? take, int? order)
```
Comment (andjiev) [inline src/MegaMall.Api/Controllers/ProductController.cs:22]: order should be enum

File: src/MegaMall.Api/Controllers/ProductController.cs [hunk 2]
```diff
@@ -19,19 +19,20 @@ public ProductController(IProductService service)
...
+        {            
             // TODO: should return mapped products
-            var products = await service.Get(page ?? 0, take ?? 10);
-            return Ok(products);
+                var products = await service.Get(page ?? 0, take ?? 10, order ?? 3);
```
Comment (andjiev) [inline src/MegaMall.Api/Controllers/ProductController.cs:25]: enum here instead of `3`

Comment (andjiev) [reply src/MegaMall.Api/Controllers/ProductController.cs:25]: also use the `CTRL + K + D` for formatting the document. There are a lot of white spaces

File: src/MegaMall.Common/Processor/SetecProcessor.cs [hunk 3]
```diff
@@ -23,7 +23,17 @@ public List<Product> ProcessPage(CrawledPage crawledPage)
                 var image = link.QuerySelector("img") as IHtmlImageElement;
                 var imageSrc = image.GetAttribute("data-echo");
                 var priceElement = (product.QuerySelector(".cena_za_kesh") as IHtmlSpanElement);
-                var price = priceElement != null && priceElement.TextContent.Length > 0 ? priceElement.TextContent.Split(" ").FirstOrDefault() : string.Empty;
+                string price = " ";
```
Comment (andjiev) [inline src/MegaMall.Common/Processor/SetecProcessor.cs:26]: `string price = string.Empty;`

File: src/MegaMall.Services/Services/IProductService.cs [hunk 4]
```diff
@@ -7,7 +7,7 @@
 
     public interface IProductService
     {
-        Task<List<Product>> Get(int? page, int? take);
+        Task<List<Product>> Get(int? page, int? take, int? order);
```
Comment (andjiev) [inline src/MegaMall.Services/Services/IProductService.cs:10]: order should be Enum here as well

File: src/MegaMall.Services/Services/ProductService.cs [hunk 5]
```diff
@@ -7,7 +7,7 @@
     using MegaMall.Common.Models;
     using MegaMall.Common.Settings;
     using MongoDB.Driver;
-
+
```
Comment (andjiev) [inline src/MegaMall.Services/Services/ProductService.cs:10]: format document

File: src/MegaMall.Services/Services/ProductService.cs [hunk 6]
```diff
@@ -21,8 +21,23 @@ public ProductService(IMegaMallDatabaseSettings settings)
             products = database.GetCollection<Product>(settings.ProductsCollectionName);
         }
 
-        public Task<List<Product>> Get(int? page, int? take) =>
-            products.Find(product => true).Skip(page * take).Limit(take).ToListAsync();
+        //public Task<List<Product>> Get(int? page, int? take) =>
+        //    products.Find(product => true).Skip(page * take).Limit(take).ToListAsync();
+        public Task<List<Product>> Get(int? page, int? take,int? order)
+        {
+            if(order == 1)
```
Comment (andjiev) [inline src/MegaMall.Services/Services/ProductService.cs:28]: replace the ifs with a switch statement for the given enum value. Also use the new switch expression from [here](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/switch-expression)

**Review by andjiev — CHANGES_REQUESTED** [review]

File: src/MegaMall.Services/Models/ProductsOrder.cs [hunk 7]
```diff
@@ -0,0 +1,9 @@
+﻿namespace MegaMall.Services.Models
+{
+    public enum OrderTypes
```
Comment (andjiev) [inline src/MegaMall.Services/Models/ProductsOrder.cs:3]: the file name should be the same

File: src/MegaMall.Services/Services/ProductService.cs [hunk 8]
```diff
@@ -21,8 +22,20 @@ public ProductService(IMegaMallDatabaseSettings settings)
             products = database.GetCollection<Product>(settings.ProductsCollectionName);
         }
 
-        public Task<List<Product>> Get(int? page, int? take) =>
-            products.Find(product => true).Skip(page * take).Limit(take).ToListAsync();
+        //public Task<List<Product>> Get(int? page, int? take) =>
```
Comment (andjiev) [inline src/MegaMall.Services/Services/ProductService.cs:25]: remove the commented code

**Review by andjiev — CHANGES_REQUESTED** [review]

File: src/MegaMall.Services/Models/ProductsOrder.cs (hunk 7 above)
Comment (DGjorgjievski) [reply src/MegaMall.Services/Models/ProductsOrder.cs:3]: OrderTypes ?

**Review by andjiev — APPROVED** [review]

**Pull request merged**